### PR TITLE
MP-7920   [BS-8103] NOX - Database configuration for "PhoneNumber"

### PR DESCRIPTION
**Pull request by becicajla**

### CODE REVIEW

File: tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs [hunk 1]
```diff
@@ -98,6 +98,7 @@ public void GeneratedEntity_Sqlite_CanSaveAndReadFields_AllTypes()
 
         var temperatureFahrenheit = 88;
         var temperaturePersistUnitAs = TemperatureTypeUnit.Celsius;
+        var phoneNumber = "38761000000";
```
Comment (dejobratic) [inline tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs:101]: Please update PostgresIntegrationTests as well.

File: src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public NoxType ForNoxType => NoxType.PhoneNumber;
+
+    public virtual bool IsDefault => true;
+
+    public void ConfigureEntityProperty(NoxSolutionCodeGeneratorState noxSolutionCodeGeneratorState, EntityTypeBuilder builder, NoxSimpleTypeDefinition property, Entity entity, bool isKey)
+    {
+        builder
+            .Property(property.Name)
+            .IsRequired(property.IsRequired)
+            .HasMaxLength(30)
```
Comment (dejobratic) [inline src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs:18]: Can we also add a validation in the PhoneNumber type for max length?

**Review by dejobratic — CHANGES_REQUESTED** [review]
Looks good. Added some comments.

File: tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs (hunk 1 above)
Comment (becicajla) [reply tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs:103]: @dejobratic  done

File: src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs (hunk 2 above)
Comment (becicajla) [reply src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs:18]: @dejobratic done

**Review by dejobratic — APPROVED** [review]

File: src/Nox.Types/Types/PhoneNumber/PhoneNumber.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ namespace Nox.Types;
 public sealed class PhoneNumber : ValueObject<string, PhoneNumber>
 {
     private static readonly Regex _phoneNumberRegex = new(@"^(?:\+\d{1,3}[\s\-]?)?(?:\(\d{1,4}\)|\d{1,4})\s*\-?\s*(?:\d{1,4}\s*\-?\s*){1,4}$", RegexOptions.Compiled, TimeSpan.FromSeconds(1));
+    private static readonly int _maxPhoneNumberLength = 30;
```
Comment (rochar) [inline src/Nox.Types/Types/PhoneNumber/PhoneNumber.cs:12]: can we make this public and reuse the value on the DatabaseConfigurator?

**Review by rochar — APPROVED** [review]
looks good, see my comment, update and you can merge it yourself

thks

File: src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            .HasConversion<PhoneNumberConverter>();
+    }
+
+    public string GetKeyPropertyName(NoxSimpleTypeDefinition key) => key.Name;
+
+    public virtual string? GetColumnType()
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs:24]: We do not need this method

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Types/PhoneNumber/PhoneNumber.cs (hunk 3 above)
Comment (becicajla) [reply src/Nox.Types/Types/PhoneNumber/PhoneNumber.cs]: resolving as done

File: src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs (hunk 4 above)
Comment (becicajla) [reply src/Nox.Types.EntityFramework/Types/PhoneNumber/PhoneNumberDatabaseConfigurator.cs]: @gjorgji-grgovski-authoritypartners I updated the PR, is it ok to merge?

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**